File: benchmark/fibonacci/fibonacci.py
# -*- coding: utf-8 -*-

import time

def fib(n):
    if n == 0:
        return 0
    if n == 1:
        return 1
    return fib(n-1)+fib(n-2) 

if __name__ == '__main__':
    N = 40
    start = time.time()
    fibonacci = fib(N)
    end = time.time()

    print("python", fibonacci, end-start, "sec.")

File: benchmark/fibonacci/Fibonacci.java
public class Fibonacci{
  public static int fib(int n){
    if(n == 0 || n == 1){
      return n;
    }else{
      return fib(n-1) + fib(n-2);
    }
  }
  public static void main(String args[]){
    int N = 40;
    long start = System.nanoTime();
    long fibonacci = fib(N);
    long time = System.nanoTime() - start;
    System.out.println("java "+fibonacci+" "+time/1000000000.+"sec.");
  }
}
